plugins/modules/user.py

```diff
@@ -127,33 +127,48 @@ def modify_user(module):
     note:
         Exits with fail_json in case of error
     return:
-        Message for successfull command
+        ( Message for command, changed status )
     '''
     attributes = module.params['attributes']
     opts = ""
     load_module_opts = None
     msg = None
+    changed = False
 
     if attributes is not None:
+        # Get all settings for user and store in dict user_attrs
+        cmd = "lsuser -C %s" % module.params['name']
+        rc, stdout, stderr = module.run_command(cmd)
+        if rc != 0:
+            msg = "\nFailed to validate attributes for the user: %s" % module.params['name']
+            module.fail_json(msg=msg, rc=rc, stdout=stdout, stderr=stderr)
+        keys = stdout.splitlines()[0].split(':')
+        values = stdout.splitlines()[1].split(':')
+        user_attrs = dict(zip(keys, values))
         for attr, val in attributes.items():
             if attr == 'load_module':
                 load_module_opts = "-R %s " % val
             else:
                 pattern = re.compile(r'(yes|true|always|no|false|never)', re.IGNORECASE)
                 if val in [True, False] or re.match(pattern, str(val)):
                     val = str(val).lower()
-                opts += "%s=\"%s\" " % (attr, val)
+                # Confirm if setting is already set before adding to opts
+                if user_attrs[attr] != val:
+                    opts += "%s=\"%s\" " % (attr, val)
+
         if load_module_opts is not None:
             opts = load_module_opts + opts
-        cmd = "chuser %s %s" % (opts, module.params['name'])
-
-        rc, stdout, stderr = module.run_command(cmd)
-
-        if rc != 0:
-            msg = "\nFailed to modify attributes for the user: %s" % module.params['name']
-            module.fail_json(msg=msg, rc=rc, stdout=stdout, stderr=stderr)
+        if opts:
+            cmd = "chuser %s %s" % (opts, module.params['name'])
+            rc, stdout, stderr = module.run_command(cmd)
+            if rc != 0:
+                msg = "\nFailed to modify attributes for the user: %s" % module.params['name']
+                module.fail_json(msg=msg, rc=rc, stdout=stdout, stderr=stderr)
+            else:
+                msg = "\nAll provided attributes for the user: %s are set SUCCESSFULLY" % module.params['name']
+            changed = True
         else:
-            msg = "\nAll provided attributes for the user: %s are set SUCCESSFULLY" % module.params['name']
+            msg = "\nNo changes were made to user: %s" % module.params['name']
 
     if module.params['password'] is not None:
         pass_msg = change_password(module)
```

Comment:
If the user also input a new password to be modified, the change password is going to return : msg, changed, a tuple instead of a plain string.
Because the variable "changed" did not exist before, the pass_msg was returned with just the message from the change_password. With the new changes, the changed variable is returned as "true". 
The side effect for this is that the pass_msg is transform from a string to a tuple,  with two elements, the message and the value "true". When the next line tries to append the pass_msg to the msg, python returns and error as you can not just add a tuple to a string.  Same problem with exist in the create_user function.

The second consideration all the attributes the user tries to change were already there (idepontency change for this PR).  In that case, if the user also input the password, and the password was successful, this means change should be true  as something was changed. 

This line needs to be something like: 
pass_msg, tmp_changed = change_password(module)
if not changed:
   changed = tmp_changed

In this way, if no value in the attributes was changed, we will set changed to whichever return we get from change_password. 

Also some of the sanity changes are not passing. We created automatic testcases that mock input and run the functions with it. It seems some of the testcases are not passing for this function. 




To test this scenario, create a playbook that modifies the attributes of a user but also input a new password with it. It should fail. 
As soon as you do this, we can help with the testcases.